content/en/docs/concepts/configuration/secret.md

```diff
@@ -15,48 +15,370 @@ weight: 30
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-basic-auth
+type: kubernetes.io/basic-auth
+stringData:
+  username: admin
+  password: t0p-Secret
+```
+
+The basic authentication Secret type is provided only for user's convenience.
+You can create an `Opaque` for credentials used for basic authentication.
+However, using the builtin Secret type helps unify the formats of your credentials
+and the API server does verify if the required keys are provided in a Secret
+configuration.
+
+#### SSH authentication Secrets
+
+The builtin type `kubernetes.io/ssh-auth` is provided for storing data used in
+SSH authentication. When using this Secret type, you will have to specify a
+`ssh-privatekey` key-value pair in the `data` (or `stringData`) field.
+as the SSH credential to use.
+
+The following YAML is an example config for a SSH authentication Secret:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-ssh-auth
+type: kubernetes.io/ssh-auth
+data:
+  ssh-privatekey: |
+     MIIEpQIBAAKCAQEAulqb/Y ... (rest of the data)
+```
+
+The SSH authentication Secret type is provided only for user's convenience.
+You can create an `Opaque` for credentials used for SSH authentication.
+However, using the builtin Secret type helps unify the formats of your credentials
+and the API server does verify if the required keys are provided in a Secret
+configuration.
+
+
+#### TLS Secrets
+
+Kubernetes provides a builtin Secret type `kubernetes.io/tls` for to storing
+information about a TLS client or server. This data is primarily used with TLS
+termination of the Ingress resource, but may be used in other types.
+When using this type of Secret, the `tls.key` and the `tls.crt` key must be provided
+in the `data` (or `stringData`) field of the Secret configuration, although the API
+server doesn't actually validate the values for each key.
+
+The following YAML contains an example config for a TLS Secret:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-tls
+type: kubernetes.io/tls
+data:
+  tls.crt: |
+    MIIC2DCCAcCgAwIBAgIBATANBgkqh ... (rest of the cert)
+  tls.key: |
+    MIIEpgIBAAKCAQEA7yn3bRHQ5FHMQ ... (rest of the key)
+```
+
+The TLS Secret type is provided for user's convenience. You can create an `Opaque`
+for credentials used for TLS server and/or client.  However, using the builtin Secret
+type helps ensure the consistency of Secret format in your project; the API server
+does verify if the required keys are provided in a Secret configuration.
+
+When creating a TLS Secret from the `kubectl` command line, you can use the
+`tls` subcommand as shown in the following example:
+
+```shell
+kubectl create secret tls my-tls-secret \
+  --cert=path/to/cert/file \
+  --key=path/to/key/file
+```
+
+The public/private key pair must exist before hand. The public key certificate
+for `--cert` must be .PEM encoded and match the given private key for `--key`.
```

Comment:
Possibly:
```suggestion
for `--cert` must be PEM encoded (Base64-encoded DER format), and match the given private key for `--key`.
```
or with more detail about the expected format:
```suggestion
for `--cert` must be PEM encoded (Base64-encoded DER format), and match the given private key for `--key`.
The private key must be in what  is commonly called PEM private key format, unencrypted.
In both cases, the initial and last lines from PEM (for example: `-----BEGIN CERTIFICATE-----` and
`-----END CERTIFICATE-----` for a certificate) are *not* included.
```

Reply:
The exclusion of the first and the last line doesn't look like something a user wants/needs to know. It is a common practice. However, including this detail doesn't hurt. I'm adding these lines.